alma/alma-woocommerce-gateway
# Add hmac verification on ipn

### Reason for change



[Linear task](https://linear.app/almapay/issue/ECOM-1821/[-wc]-add-hmac-verification-on-ipn)

Need to implement the verification of HMAC signature on IPN.

### Code changes



- Add a securityHelper to call php-client's function to check HMAC signature.
- Call this helper in the PaymentHelper.
- Add some unit tests.

### How to test

_As a reviewer, you are encouraged to test the PR locally._


You can run unit test locally.

[QA ENV](https://qa-wc-hmac-woocommerce-9-0-2.integrations.getalma.eu/)

Create a payment and check on Datadog if the log of IPN is ok.
You can try to use Postman to call the IPN url with no signature and bad signature and check the error message.

### Checklist for authors and reviewers




### Non applicable
